security/advancedtls/examples/credential_reloading_from_files/client/main.go

```diff
@@ -81,18 +99,55 @@ func main() {
 		log.Fatalf("advancedtls.NewClientCreds(%v) failed: %v", options, err)
 	}
 
-	// At initialization, the connection should be good.
+	// Make a connection using the old credentials.
 	ctx, cancel := context.WithTimeout(context.Background(), defaultConnTimeout)
 	defer cancel()
-	conn, err := grpc.DialContext(ctx, address, grpc.WithTransportCredentials(clientTLSCreds))
+	oldConn, err := grpc.DialContext(ctx, address, grpc.WithTransportCredentials(clientTLSCreds))
 	if err != nil {
 		log.Fatalf("grpc.DialContext to %s failed: %v", address, err)
 	}
-	greetClient := pb.NewGreeterClient(conn)
-	reply, err := greetClient.SayHello(ctx, &pb.HelloRequest{Name: "gRPC"}, grpc.WaitForReady(true))
+	oldClient := pb.NewGreeterClient(oldConn)
+	_, err = oldClient.SayHello(ctx, &pb.HelloRequest{Name: "gRPC"}, grpc.WaitForReady(true))
+	if err != nil {
+		log.Fatalf("oldClient.SayHello failed: %v", err)
+	}
+	defer oldConn.Close()
+
+	// Change the contents of the credential files.
+	// Cert "another_client_cert_1.pem" is also trusted by "server_cert_1.pem".
+	if err := copyFileContents(testdata.Path("another_client_cert_1.pem"), certTmpFile.Name()); err != nil {
```

Comment:
This isn't exactly what I had in mind -- IIUC the credentials updating would typically happen externally, not within the application.  So this code to do that clutters the example showing users how to write their application.

Maybe move the credentials-overwriting parts into the script instead of the client?  The client could just perform RPCs slowly (with a new connection each time), and we should eventually see that the new credentials are being used, shortly after overwriting them.

Reply:
SG! Moving the credential-rewriting parts to the script is a great idea! I've updated the code to behave that way. Could you please take a look again? Thank you so much!